.circleci/config.yml

```diff
@@ -53,26 +146,122 @@ unit-tests-defaults: &unit-tests-defaults
   executor: golang
   resource_class: medium+
 
+# The default circle ci checkout does a full checkout and that brings about
+# 450M of history that we don't need.
+shallow-checkout: &shallow-checkout
+  run:
+    name: shallow checkout
+    command: |
+      mkdir ~/.ssh && ssh-keyscan github.com >> ~/.ssh/known_hosts
+      git clone --quiet --depth 1 "$CIRCLE_REPOSITORY_URL" --branch "$CIRCLE_BRANCH" ~/repos/geth
+
+install-go: &install-go
+  run:
+    name: install go
+    command: |
+      # Install go so the e2e tests can build geth
+      mkdir -p ~/repos/golang
+      wget https://dl.google.com/go/go1.16.4.linux-amd64.tar.gz
+      tar xf go1.16.4.linux-amd64.tar.gz -C ~/repos/golang
+      ~/repos/golang/go/bin/go version
+
+save-go-cache: &save-go-cache
+  save_cache:
+    key: go-cache-<<pipeline.parameters.go-cache-version>>-{{ checksum "~/repos/geth/go.sum" }}-{{ checksum "~/repos/month" }}
+    paths:
+      - ~/repos/gocache
+
+restore-go-cache: &restore-go-cache
+  restore_cache:
+    keys:
+      - go-cache-<<pipeline.parameters.go-cache-version>>-{{ checksum "~/repos/geth/go.sum" }}-{{ checksum "~/repos/month" }}
+
+go-mod-cache: &save-go-mod-cache
+  save_cache:
+    key: go-mod-<<pipeline.parameters.go-cache-version>>-{{ checksum "~/repos/geth/go.sum" }}
+    paths:
+      - ~/repos/gocache
+
+restore-go-mod-cache: &restore-go-mod-cache
+  restore_cache:
+    keys:
+      - go-mod-<<pipeline.parameters.go-cache-version>>-{{ checksum "~/repos/geth/go.sum" }}
+
+store-month-in-file: &store-month-in-file
```

Comment:
https://circleci.com/docs/2.0/caching/#introduction
max caching storage duration is 15 days

Reply:
Hmmm, I read that as meaning that caches would persist for 15 days after their last use. There doesn't seem to be any sense in deleting a cache if it's in constant use, but their documentation seems to suggest otherwise. I've opened an [issue](https://github.com/circleci/circleci-docs/issues/6909) to get some clarification on that, in the meantime I think we can keep this as it will ensure periodic cache changes and it's not a problem if the cache changes slightly more often.

Reply:
Ok I discovered that we have some old caches [this job restores a cache that is over 90 days old](https://app.circleci.com/pipelines/github/celo-org/celo-blockchain/7979/workflows/a76c3667-d19b-428d-979a-c2c6d6a3b35c/jobs/82940) so I think we need to keep the rotation functionlaity.